luigi/contrib/snowflake.py

```diff
@@ -0,0 +1,523 @@
+    """Abstractable SnowflakeAuth base class that can be replace by tool
+    specific implementation."""
+    @property
+    def configuration_section(self):
+        """
+        Override to change the configuration section used
+        to obtain default credentials.
+        """
+        return 'snowflake'
+
+    @abc.abstractmethod
+    def credentials(self, task):
+        pass
+
+
+class _SettingsMixins(object):
+    """
+    Generic mixins that defines how to fetch the database and warehouse
+    configuration for a given snowflake integration.
+    """
+    @property
+    def configuration_section(self):
+        """
+        Override to change the configuration section used
+        to obtain default credentials.
+        """
+        return 'snowflake'
+
+    @property
+    def warehouse(self):
+        """
+        Fetch the warehouse configuration from the luigi configuration file under:
+
+        [snowflake]
+        warehouse=value
+        """
+        return luigi.configuration.get_config().get(self.configuration_section, 'warehouse', default=None)
+
+    @property
+    def role(self):
+        """
+        Fetch the role configuration from the luigi configuration file under:
+
+        [snowflake]
+        role=value
+        """
+        return luigi.configuration.get_config().get(self.configuration_section, 'role', default=None)
+
+
+class AwsKeyAuth(SnowflakeAuth):
+    def credentials(self):
+        """
+        Return an AWS credential string for the provided task.
+
+        If no valid credentials are set, raise a NotImplementedError.
+        """
+        if self.aws_access_key_id and self.aws_secret_access_key:
+            return "AWS_KEY_ID='{key}' AWS_SECRET_KEY='{secret}'".format(
+                key=self.aws_access_key_id,
+                secret=self.aws_secret_access_key)
+        else:
+            raise NotImplementedError("Missing Credentials. "
+                                      "Ensure that the auth args below are set "
+                                      "in a configuration file, environment variables or by "
+                                      "being overridden in the task: "
+                                      "'aws_access_key_id' AND 'aws_secret_access_key'")
+
+    @property
+    def aws_access_key_id(self):
+        """
+        Override to return the key id.
+        """
+        return luigi.configuration.get_config().get(self.configuration_section, 'aws_access_key_id')
+
+    @property
+    def aws_secret_access_key(self):
+        """
+        Override to return the secret access key.
+        """
+        return luigi.configuration.get_config().get(self.configuration_section, 'aws_secret_access_key')
+
+
+class SnowflakeTarget(postgres.PostgresTarget):
+    """
+    Target for a resource in Snowflake.
+
+    Snowflake connection string is similar to postgres with a few adjustments.
+    """
+    marker_table = luigi.configuration.get_config().get(
+        'snowflake',
```

Comment:
just a note.... at this point, we've allowed the section to be configured for settings and auth, but are hardcoding the db section here. I'd suspect someone would likely store all these configuration variables in the same cfg section.

Do we require they all exist in `snowflake`? Allow them to be individually configurable? Or set once, use in all places?